### ISSUE TITLE
Custom filters do not work consistenly (Windows works, linux fails)

### ISSUE THREAD

**Issue by Mansikka** [etianen/django-python3-ldap#191]
Using this in Django project, we are using AD to bring in users. However, we have two OUs we want users to come from. Let's call them office1 and office2.

Issue is that we want to _only_ bring people from those offices, not any of our remote desktops or others. Using the tutorials provided, we have done following:

In settings.py
```
from project.ldap_filter import custom_format_search_filters
LDAP_AUTH_SEARCH_BASE = "DC=dc1,DC=dc2"
# Custom setting
LDAP_AUTH_FORMAT_SEARCH_FILTERS = custom_format_search_filters
```

and in our ldap_filter.py:
```
from django_python3_ldap.utils import format_search_filters

def custom_format_search_filters(ldap_fields ) -> list:
    """
    Custom search so we include hollming in our search

    ### Args:
    - ldap_fields ([type]): LDAP fields

    ### Returns:
    - list: Custom filtering
    """
    # Call the base format callable.
    search_filters = format_search_filters(ldap_fields)
    # Advanced: apply custom LDAP filter logic.
    search_filters.append("(|(ou=office1, ou=office2))")
    # All done!
    return search_filters

```
By all what we have read, this shoulf give us entries in office1 and office 2 and ignore all others. Yet, if we run this, it just gives all.

If we change
`LDAP_AUTH_SEARCH_BASE = "DC=dc1,DC=dcd2"`
to`
LDAP_AUTH_SEARCH_BASE = "OU=office1,DC=dc1,DC=dc2"`

It works on Windows, on my local server, but when we move to our development server (Linux) it just looks for users in office1.

If we put office2 there, it just searches office2 in both cases. Why is this? What are we doing wrong? Both offices are on the same level in the hierarchy.

**Comment by etianen** [comment]
Are your local server and development server both using the same
LDAP_AUTH_URL? And does this resolve to the same LDAP server?

**Comment by Mansikka** [comment]
Yes, they are identical in their setups. Tinkering around I found recently that it seems to just not find OUs at all.

**Comment by etianen** [comment]
That's really odd. There's nothing django-python3-ldap does that is
os-specific. Possibly it's:

   - Some underlying problem in some compiled python extension in the
   underlying ldap lib.
   - Some networking issue manifesting as a "no results found".

I think you'll need to debug this one, as it seems specific to your setup.
I'll take an MR to fix any issues you find.

**Comment by Mansikka** [comment]
Thanks, we managed to debug it to certain degree. It still refuses to look through two OUs, but we managed to find some workarounds.

One possible issue, that we recently found, was that our test server is using Python 3.6.X, while development is done in Python 3,7 enviroment.

**Comment by etianen** [comment]
Interesting. I can't see why that would be an issue, unless it affects some
underlying lib.

**Comment by Mansikka** [comment]
At this point we are kinda treating whole system (especially LDAP connection to AD) as a black magic. I'll drop info if we find anything new.
